src/test/architecture/test_graph_search.py

```diff
@@ -591,3 +688,296 @@ def test_single_node_cyclic_case(self, mocker):
             (node_path_mock, comm_mock, node_path_mock),
         )
         assert [v.to_value() for v in paths] == [expect.to_value()]
+
+    def test_to_path_construction_order(self, mocker):
```

Comment:
It is better to change this function to focus on testing `to_path` with variation of construction_order, since currently this function is difficult to understand because of its complexity. My suggestions are:
- moving test of `CommunicationStruct` to new file dedicated to testing `CommunicationStruct`
- moving test of `_find_comm` to new function `test_find_comm`
- moving test of `_find_node_path` to new function `test_find_node_path`
- moving test of `__init__` to new function `test_init`. I think test of `__init__` may not be necessary. If so, just removing test of `__init__` is good.

Reply:
I split the part you pointed out.